plugins/inlineStyles.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+exports.type   = 'full';
+
+exports.active = true;
+
+exports.params = {
+  onlyMatchedOnce: true
+};
+
+exports.description = 'inline styles (optionally skip selectors that match more than once)';
+
+
+var cheerioSupport = require('../lib/svgo/cheerio-support'),
+    SPECIFICITY    = require('specificity'),
+    stable         = require('stable'),
+    csso           = require('csso');
+
+
+/**
+  * Moves styles from <style> to element styles
+  *
+  * @author strarsis <strarsis@gmail.com>
+  */
+exports.fn = function(data, opts) {
+
+  // svgo ast to cheerio ast
+  var $ = cheerioSupport.svgoAst2CheerioAst(data);
+
+  var $styles       = $('style');
+  var styleItems    = [];
+  var selectorItems = [];
+  $styles.each(function(si, $style) {
+    if($style.children.length == 0) {
+      return;
+    }
+
+    var cssStr = $style.children[0].data;
+    if(cssStr.length == 0) {
+      return;
+    }
+
+    var cssAst = csso.parse(cssStr, {
+                   context: 'stylesheet'
+                 });
+    styleItems.push({
+      $style: $style,
+      cssAst: cssAst
+    });
+
+    csso.walk(cssAst, function(node, item) {
+      // single selector
+      if(node.type === 'SimpleSelector') {
+        var selectorStr  = csso.translate(node);
+        var selectorItem = {
+          selectorStr:        selectorStr,
+          simpleSelectorItem: item,
+          rulesetNode:        this.ruleset
+        };
+        selectorItems.push(selectorItem);
+      }
+    });
+  });
+
+  // stable sort selectors by specificity
+  var selectorItemsSorted = stable(selectorItems, function(item1, item2) {
+    return SPECIFICITY.compare(item1.selectorStr, item2.selectorStr);
+  });
+
+  for(var selectorItem of selectorItemsSorted) {
```

Comment:
Unfortunatetly, `for-of` loops aren't supported in Node 0.10 which is still supported. However, probably not for a long time.